gradio/route_utils.py

```diff
@@ -583,3 +584,54 @@ def starts_with_protocol(string: str) -> bool:
     """
     pattern = r"^[a-zA-Z][a-zA-Z0-9+\-.]*://"
     return re.match(pattern, string) is not None
+
+
+def get_core_url(original_url: str) -> str:
+    """
+    Returns the core url of a given url. If the url does not have a scheme, it is assumed to be http.
+    Examples:
+        get_core_url("https://www.gradio.app") -> "www.gradio.app"
+        get_core_url("localhost:7860") -> "localhost"
+        get_core_url("127.0.0.1") -> "127.0.0.1"
+    """
+    if not original_url:
+        return ""
+    if "://" not in original_url:
+        original_url = "http://" + original_url
+    return httpx.URL(original_url).host
```

Comment:
nit: I prefer a standard library `urllib` rather than `httpx`:
```python
from urllib.parse import urlparse
...
urlparse(original_url).hostname
```

Reply:
Sure yeah I can replace